sklearn/metrics/_classification.py

```diff
@@ -353,7 +354,10 @@ def accuracy_score(y_true, y_pred, *, normalize=True, sample_weight=None):
     >>> accuracy_score(np.array([[0, 1], [1, 1]]), np.ones((2, 2)))
     0.5
     """
-    xp, _, device = get_namespace_and_device(y_true, y_pred, sample_weight)
+    xp, _, device = get_namespace_and_device(y_pred)
+    y_true, sample_weight = _convert_to_reference(
+        reference=y_pred, arrays=(y_true, sample_weight)
+    )
```

Comment:
I agree that returning `xp` and `device` from `convert_to_reference` feels like too much. But it also feels weird to have this "duplication" of `get_namespace_and_device` and `convert_to_reference`. The only alternative I can think of, though I'm not sure I like it, is to have `convert_to_reference(arrays, namespace=xp, device=device)` (but then maybe rename it to `move_to` or some such.